src/Tizen.Uix.Stt/Tizen.Uix.Stt/SttClient.cs

```diff
@@ -504,88 +567,132 @@ public SttClient()
                     }
 
                     _defaultLanguageChanged -= value;
+                    if (null == _defaultLanguageChanged)
+                    {
+                        SttError error = SttUnsetDefaultLanguageChangedCB(_handle);
+                        if (error != SttError.None)
+                        {
+                            Log.Error(LogTag, "Remove DefaultLanguageChanged Failed with error " + error);
+                            throw ExceptionFactory.CreateException(error);
+                        }
+                    }
                 }
             }
-
         }
 
         /// <summary>
         /// Event to be invoked to detect the engine change.
         /// </summary>
         /// <since_tizen> 3 </since_tizen>
+        /// <privilege>http://tizen.org/privilege/recorder</privilege>
+        /// <exception cref="InvalidOperationException">This exception can be due to an invalid state.</exception>
+        /// <exception cref="NotSupportedException">This exception can be due to STT not supported.</exception>
+        /// <exception cref="UnauthorizedAccessException">This exception can be due to permission denied.</exception>
+        /// <exception cref="ArgumentException">This exception can be due to improper value provided while setting the value.</exception>
+        /// <pre>
+        /// The state must be created.
+        /// </pre>
         public event EventHandler<EngineChangedEventArgs> EngineChanged
         {
             add
             {
                 lock (thisLock)
                 {
-                    _engineDelegate = (IntPtr handle, IntPtr engineId, IntPtr language, bool supportSilence, bool needCredential, IntPtr userData) =>
-                    {
-                        string engineIdString = Marshal.PtrToStringAnsi(engineId);
-                        string languageString = Marshal.PtrToStringAnsi(language);
-                        EngineChangedEventArgs args = new EngineChangedEventArgs(engineIdString, languageString, supportSilence, needCredential);
-                        _engineChanged?.Invoke(this, args);
-                    };
-                    SttError error = SttSetEngineChangedCB(_handle, _engineDelegate, IntPtr.Zero);
-                    if (error != SttError.None)
+                    if (State.Created != CurrentState)
                     {
-                        Log.Error(LogTag, "Add EngineChanged Failed with error " + error);
+                        Log.Error(LogTag, "Add EngineChanged Failed with error " + SttError.InvalidState);
+                        throw ExceptionFactory.CreateException(SttError.InvalidState);
                     }
-                    else
+                    if (null == _engineChanged)
                     {
-                        _engineChanged += value;
+                        _engineDelegate = (IntPtr handle, IntPtr engineId, IntPtr language, bool supportSilence, bool needCredential, IntPtr userData) =>
```

Comment:
If change the type of parameter from IntPtr to String, manual marshaling may be not need.